src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -112,6 +127,25 @@ private void EndRequest(HttpContext httpContext, long timestamp)
                     telemetry.Name = httpContext.Request.Method + " " + httpContext.Request.Path.Value;
                 }
 
+                if (string.IsNullOrEmpty(telemetry.Context.InstrumentationKey))
```

Comment:
Should this (setting the iKey based on request headers) be done in a TelemetryInitializer? That way, any other telemetry associated with the request (e.g. Exceptions, Event, etc.) sent prior to OnEndRequest will also be tagged with the same iKey. Or is that not required/wanted here?